Mathlib/Analysis/CStarAlgebra/CStarMatrix.lean

```diff
@@ -0,0 +1,680 @@
+(which takes the norm of the biggest entry as the norm of the matrix)
+in order to show that the map `ofMatrix` is bilipschitz. We then finally register the
+`NormedAddCommGroup (C⋆ᵐᵒᵈ (n → A))` instance via `NormedAddCommGroup.ofCoreReplaceAll`.
+-/
+
+namespace CStarMatrix
+
+variable {m n A : Type*} [Fintype m] [Fintype n] [DecidableEq n]
+  [NonUnitalCStarAlgebra A] [PartialOrder A] [StarOrderedRing A]
+
+private noncomputable def normedAddCommGroupAux : NormedAddCommGroup (CStarMatrix m n A) :=
+  .ofCore CStarMatrix.normedSpaceCore
+
+attribute [local instance] normedAddCommGroupAux
+
+private def normedSpaceAux : NormedSpace ℂ (CStarMatrix m n A) :=
+  .ofCore CStarMatrix.normedSpaceCore
+
+/- In this `Aux` section, we locally activate the following instances: a norm on `CStarMatrix`
+ which induces a topology that is not defeq with the matrix one, and the elementwise norm on
+ matrices, in order to show that the two topologies are in fact equal -/
+attribute [local instance] normedSpaceAux Matrix.normedAddCommGroup Matrix.normedSpace
+
+private lemma nnnorm_le_of_forall_inner_le {M : CStarMatrix m n A} {C : ℝ≥0}
+    (h : ∀ v w, ‖⟪w, CStarMatrix.toCLM M v⟫_A‖₊ ≤ C * ‖v‖₊ * ‖w‖₊) : ‖M‖₊ ≤ C :=
+  CStarMatrix.norm_le_of_forall_inner_le fun v w => h v w
+
+open Finset in
+private lemma lipschitzWith_toMatrixAux [DecidableEq m] :
+    LipschitzWith 1 (ofMatrixₗ.symm (R := ℂ) : CStarMatrix m n A → Matrix m n A) := by
+  refine AddMonoidHomClass.lipschitz_of_bound_nnnorm _ _ fun M => ?_
+  rw [one_mul, ← NNReal.coe_le_coe, coe_nnnorm, coe_nnnorm, Matrix.norm_le_iff (norm_nonneg _)]
+  exact fun _ _ ↦ CStarMatrix.norm_entry_le_norm
+
+open CStarMatrix WithCStarModule in
+private lemma antilipschitzWith_toMatrixAux :
+    AntilipschitzWith (Fintype.card m * Fintype.card n)
+      (ofMatrixₗ.symm (R := ℂ) : CStarMatrix m n A → Matrix m n A) := by
+  refine AddMonoidHomClass.antilipschitz_of_bound _ fun M => ?_
+  calc
+    ‖M‖ ≤ ∑ i, ∑ j, ‖M i j‖ := by
+      rw [norm_def]
+      refine (toCLM M).opNorm_le_bound (by positivity) fun v => ?_
+      simp only [toCLM_apply_eq_sum, equiv_symm_pi_apply, Finset.sum_mul]
+      apply pi_norm_le_sum_norm _ |>.trans
+      gcongr with i _
+      apply norm_sum_le _ _ |>.trans
+      gcongr with j _
+      apply norm_mul_le _ _ |>.trans
+      gcongr
+      exact norm_apply_le_norm v j
+    _ ≤ ∑ _ : m, ∑ _ : n, ‖ofMatrixₗ.symm (R := ℂ) M‖ := by
+      gcongr with i _ j _
+      exact ofMatrixₗ.symm (R := ℂ) M |>.norm_entry_le_entrywise_sup_norm
+    _ = _ := by simp [mul_assoc]
+
+private lemma uniformInducing_toMatrixAux [DecidableEq m] :
+    IsUniformInducing (ofMatrix.symm : CStarMatrix m n A → Matrix m n A) :=
+  AntilipschitzWith.isUniformInducing antilipschitzWith_toMatrixAux
+    lipschitzWith_toMatrixAux.uniformContinuous
+
+private lemma uniformity_eq_aux [DecidableEq m] :
+    𝓤 (CStarMatrix m n A) = (𝓤[Pi.uniformSpace _] :
+      Filter (CStarMatrix m n A × CStarMatrix m n A)) := by
```

Comment:
I think you likely shouldn't even need this lemma and the next one (or else you can make the proofs much easier) if you use the results in https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/MetricSpace/Bilipschitz.html

Reply:
That would entail using `comap` to define the topology, uniformity and bornology on `CStarMatrix` instead of just using `Pi.topologicalSpace` and so on as is done for `Matrix`, `piLp` etc. We've had this debate before, but I still feel that this pattern is so prevalent in mathlib that we would've noticed by now if this were a major problem.

Reply:
Oh that's right, I forgot. Sorry